.dotnet/src/Custom/Chat/ChatResponseFormat.cs

```diff
@@ -18,28 +20,49 @@ namespace OpenAI.Chat;
 /// the model.
 /// </para>
 /// </remarks>
-[CodeGenModel("CreateChatCompletionRequestResponseFormat")]
-public partial class ChatResponseFormat
+[CodeGenModel("ChatResponseFormat")]
+public abstract partial class ChatResponseFormat
 {
-    // CUSTOM: Made internal.
-
-    /// <summary> Must be one of `text` or `json_object`. </summary>
-    [CodeGenMember("Type")]
-    internal InternalCreateChatCompletionRequestResponseFormatType? Type { get; set; }
+    /// <summary> text. </summary>
+    public static ChatResponseFormat Text { get; } = new InternalChatResponseFormatText();
+    /// <summary> json_object. </summary>
+    public static ChatResponseFormat JsonObject { get; } = new InternalChatResponseFormatJsonObject();
 
-    // CUSTOM: Made internal.
-    /// <summary> Initializes a new instance of <see cref="ChatResponseFormat"/>. </summary>
-    internal ChatResponseFormat()
+    public static ChatResponseFormat FromJsonSchema(
+        string name,
+        string description = null,
+        BinaryData jsonSchema = null,
+        bool? useStrictSchema = null)
     {
+        InternalResponseFormatJsonSchemaJsonSchema internalSchema = new(
+            description,
+            name,
+            jsonSchema,
+            useStrictSchema,
+            null);
+        return new InternalChatResponseFormatJsonSchema(internalSchema);
     }
 
-    internal ChatResponseFormat(InternalCreateChatCompletionRequestResponseFormatType? type)
+    [EditorBrowsable(EditorBrowsableState.Never)]
+    public override bool Equals(object obj)
     {
-        Type = type;
+        return
+            (this is InternalChatResponseFormatText && obj is InternalChatResponseFormatText)
+            || (this is InternalChatResponseFormatJsonObject && obj is InternalChatResponseFormatJsonObject)
+            || (this is InternalChatResponseFormatJsonSchema thisSchema && obj is InternalChatResponseFormatJsonSchema otherSchema && thisSchema.JsonSchema.Name == otherSchema.JsonSchema.Name);
     }
 
-    /// <summary> text. </summary>
-    public static ChatResponseFormat Text { get; } = new ChatResponseFormat(InternalCreateChatCompletionRequestResponseFormatType.Text);
-    /// <summary> json_object. </summary>
-    public static ChatResponseFormat JsonObject { get; } = new ChatResponseFormat(InternalCreateChatCompletionRequestResponseFormatType.JsonObject);
+    [EditorBrowsable(EditorBrowsableState.Never)]
+    public static bool operator==(ChatResponseFormat first, ChatResponseFormat second)
+        => first.Equals(second);
+
+    [EditorBrowsable(EditorBrowsableState.Never)]
+    public static bool operator!=(ChatResponseFormat first, ChatResponseFormat second)
+        => !first.Equals(second);
+
+    [EditorBrowsable(EditorBrowsableState.Never)]
+    public override int GetHashCode()
+    {
+        return base.GetHashCode();
+    }
```

Comment:
Honest answer: I'm not sure.

On one hand, I can't imagine a practical case where you'd want to be able to check the response format via a comparison like this; on the other, I know there's a lot I can't imagine.

Given "we can add it later," would you prefer to remove them all for now?

Reply:
Removing them for now and updating the smoke test to just check MRW serialization.

Reply:
I agree to remove them for now.

Reply:
Ended up needing to keep a basic subset to avoid unnecessary breaking changes and keep comparison reasonable. Let me know if you have any concerns with the comparison logic; it's more code than I'd like, but I do think it enables the key comparisons to work.